src/transformers/pipelines/__init__.py

```diff
@@ -661,12 +665,34 @@ def pipeline(
     load_tokenizer = type(model_config) in TOKENIZER_MAPPING or model_config.tokenizer_class is not None
     load_feature_extractor = type(model_config) in FEATURE_EXTRACTOR_MAPPING or feature_extractor is not None
 
+    if (
+        tokenizer is None
+        and not load_tokenizer
+        and normalized_task not in NO_TOKENIZER_TASKS
+        and type(model_config) in MULTI_MODEL_CONFIGS
```

Comment:
Here we could use `model_config.model_type` to avoid importing the config classes, or use `model_config.__class__.___name__` to compare with strings instead of objects.